drivers/adc/adc_ltc2451.c

```diff
@@ -0,0 +1,102 @@
+/* LLTC LTC2451 ADC
+ *
+ * Copyright (c) 2023 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ltc2451, CONFIG_ADC_LOG_LEVEL);
+
+#define DT_DRV_COMPAT lltc_ltc2451
+
+struct ltc2451_config {
+	struct i2c_dt_spec i2c;
+	uint8_t conversion_speed;
+};
+
+static int ltc2451_channel_setup(const struct device *dev,
+				const struct adc_channel_cfg *channel_cfg)
+{
+	(void)dev;
+	(void)channel_cfg;
+	return 0; /* there is only 1 channel which requires no real setup */
```

Comment:
Return an errno for an invalid channel id

Reply:
Just return an error for a channel greater than 1 you mean? There's only 1 channel and it doesn't require any particular setup. It doesn't really even need to be called tbh.

Reply:
I was thinking something like: 
`	if (channel_cfg->channel_id != 0) {
		LOG_ERR("unsupported channel id '%d'", channel_cfg->channel_id);
		return -ENOTSUP;
	}`

Reply:
Has now been changed